homeassistant/components/zha/siren.py

```diff
@@ -0,0 +1,135 @@
+"""Support for ZHA sirens."""
+
+from __future__ import annotations
+
+import functools
+from typing import Any
+
+from homeassistant.components.siren import (
+    ATTR_DURATION,
+    DOMAIN,
+    SUPPORT_DURATION,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SirenEntity,
+)
+from homeassistant.components.siren.const import (
+    ATTR_TONE,
+    ATTR_VOLUME_LEVEL,
+    SUPPORT_TONES,
+    SUPPORT_VOLUME_SET,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+
+from .core import discovery
+from .core.channels.security import IasWd
+from .core.const import (
+    CHANNEL_IAS_WD,
+    DATA_ZHA,
+    DATA_ZHA_DISPATCHERS,
+    SIGNAL_ADD_ENTITIES,
+    WARNING_DEVICE_MODE_BURGLAR,
+    WARNING_DEVICE_MODE_EMERGENCY,
+    WARNING_DEVICE_MODE_EMERGENCY_PANIC,
+    WARNING_DEVICE_MODE_FIRE,
+    WARNING_DEVICE_MODE_FIRE_PANIC,
+    WARNING_DEVICE_MODE_POLICE_PANIC,
+    WARNING_DEVICE_MODE_STOP,
+    WARNING_DEVICE_SOUND_HIGH,
+    WARNING_DEVICE_STROBE_NO,
+)
+from .core.registries import ZHA_ENTITIES
+from .core.typing import ChannelType, ZhaDeviceType
+from .entity import ZhaEntity
+
+STRICT_MATCH = functools.partial(ZHA_ENTITIES.strict_match, DOMAIN)
+FIVE_SECONDS = 5
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zigbee Home Automation siren from config entry."""
+    entities_to_create = hass.data[DATA_ZHA][DOMAIN]
+
+    unsub = async_dispatcher_connect(
+        hass,
+        SIGNAL_ADD_ENTITIES,
+        functools.partial(
+            discovery.async_add_entities,
+            async_add_entities,
+            entities_to_create,
+            update_before_add=False,
+        ),
+    )
+    hass.data[DATA_ZHA][DATA_ZHA_DISPATCHERS].append(unsub)
+
+
+@STRICT_MATCH(channel_names=CHANNEL_IAS_WD)
+class ZHASiren(ZhaEntity, SirenEntity):
+    """Representation of a ZHA siren."""
+
+    def __init__(
+        self,
+        unique_id: str,
+        zha_device: ZhaDeviceType,
+        channels: list[ChannelType],
+        **kwargs,
+    ) -> None:
+        """Init this sensor."""
+        super().__init__(unique_id, zha_device, channels, **kwargs)
+        self._channel: IasWd = channels[0]
+        self._attr_supported_features = (
+            SUPPORT_TURN_ON
+            | SUPPORT_TURN_OFF
+            | SUPPORT_DURATION
+            | SUPPORT_VOLUME_SET
+            | SUPPORT_TONES
+        )
+        self._attr_is_on: bool = False
+        self._attr_available_tones: list[int | str] | dict[int, str] | None = {
```

Comment:
All sirens have the same available tones ?

Reply:
I'll double check the spec... but I didn't see anything that made it seem otherwise.